### PR TITLE
Added table `file`

### PR DESCRIPTION
**Pull request by nulls**
### What's done:
- added table `file`
- supported storing information in database for uploaded file
- added migration from old storage to new storage

It's part of https://github.com/saveourtool/save-cloud/pull/1670

### CODE REVIEW

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/DownloadFilesController.kt [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package com.saveourtool.save.backend.controllers
 
 import com.saveourtool.save.agent.TestExecutionDto
+import com.saveourtool.save.backend.service.*
```
Comment (nulls) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/DownloadFilesController.kt:4]: can be removed

File: gradle/plugins/src/main/kotlin/com/saveourtool/save/buildutils/save-agent-configuration.gradle.kts [hunk 2]
```diff
@@ -18,7 +18,7 @@ plugins {
 val downloadSaveAgentDistroTaskProvider: TaskProvider<Download> = tasks.register<Download>("downloadSaveAgentDistro") {
     enabled = findProperty("saveAgentDistroFilepath") != null
 
-    val saveAgentDistroFilepath = findProperty("saveAgentDistroFilepath")?.toString() ?: "file:\\\\"
+    val saveAgentDistroFilepath = findProperty("saveAgentDistroFilepath")?.toString() ?: "file:\\\\not-found"
```
Comment (nulls) [inline gradle/plugins/src/main/kotlin/com/saveourtool/save/buildutils/save-agent-configuration.gradle.kts:21]: it does resolve IDEA import issue (one of them)

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/service/ExecutionService.kt [hunk 3]
```diff
@@ -35,6 +34,8 @@ class ExecutionService(
     private val lnkContestProjectService: LnkContestProjectService,
     private val lnkContestExecutionService: LnkContestExecutionService,
     private val lnkExecutionTestSuiteService: LnkExecutionTestSuiteService,
+    @Lazy private val agentService: AgentService,
```
Comment (nulls) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/service/ExecutionService.kt:37]: need to revert

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/storage/FileStorage.kt [hunk 4]
```diff
@@ -1,54 +1,55 @@
...
+import java.time.Instant
+import javax.annotation.PostConstruct
+
+import kotlinx.datetime.*
 
 /**
  * Storage for evaluated tools are loaded by users
  */
 @Service
 class FileStorage(
```
Comment (nulls) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/storage/FileStorage.kt:27]: let's rename the class name for it to new one

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/DownloadFilesController.kt (hunk 1 above)
Comment (nulls) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/DownloadFilesController.kt]: done

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/storage/FileStorage.kt [hunk 5]
```diff
@@ -5,7 +5,7 @@ import com.saveourtool.save.domain.FileInfo
 import com.saveourtool.save.domain.FileKey
 import com.saveourtool.save.domain.ProjectCoordinates
 import com.saveourtool.save.storage.AbstractFileBasedStorage
-import com.saveourtool.save.utils.pathNamesTill
```
Comment (nulls) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/storage/FileStorage.kt:8]: can be rollbacked

Comment (nulls) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/storage/FileStorage.kt]: done

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/storage/FileStorage.kt (hunk 4 above)
Comment (nulls) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/storage/FileStorage.kt]: done

File: db/v-2/tables/file.xml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+    <changeSet id="file-1" author="nulls" context="dev or prod">
+        <createTable tableName="file">
+            <column name="id" type="bigint" autoIncrement="true">
+                <constraints primaryKey="true" nullable="false"/>
+            </column>
+            <column name="name" type="varchar(100)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="uploaded_time" type="varchar(100)">
```
[the remark this answers is not in the document]
Comment (nulls) [reply db/v-2/tables/file.xml:16]: yes, should be

**Review by Cheshiriks — APPROVED** [review]

**Pull request merged**